ipa-core/src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -169,7 +169,11 @@ where
         )
         .await?;
         match (&mut y, y_bit) {
-            (Some(y), Some(y_bit)) => y.push(y_bit),
+            (Some(y), Some(y_bit)) => {
+                // TODO: this should be replaced with real malicious security.
```

Comment:
@danielmasny - what is appropriate to do here?

Reply:
Yes we need to change the code here. We need to run the integer_add and then validate. Only afterwards we can reveal. So this is a place where the input context should be an upgradable context such that we can upgrade it to get a validator, then we run integer_add on the upgraded context, we call validate afterwards we can drop the validator and upgraded context. Then we can use the upgradable context for the reveal. We still need to discuss how to handle safety properly. For example we do not want to allow calling reveal when there exists any unvalidated multiplication.